modules/apps/object/object-api/src/main/java/com/liferay/object/util/ObjectScriptVariablesUtil.java

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.object.util;
+
+import com.liferay.object.model.ObjectDefinition;
+import com.liferay.object.model.ObjectField;
+import com.liferay.object.service.ObjectFieldLocalServiceUtil;
+import com.liferay.object.system.SystemObjectDefinitionMetadata;
+import com.liferay.object.system.SystemObjectDefinitionMetadataTracker;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+import com.liferay.portal.vulcan.dto.converter.DTOConverter;
+import com.liferay.portal.vulcan.dto.converter.DTOConverterRegistry;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author Selton Guedes
+ */
+public class ObjectScriptVariablesUtil {
+
+	public static Map<String, Object> toVariables(
+		DTOConverterRegistry dtoConverterRegistry,
+		ObjectDefinition objectDefinition, JSONObject payloadJSONObject,
+		SystemObjectDefinitionMetadataTracker
+			systemObjectDefinitionMetadataTracker) {
+
+		Map<String, Object> allowedVariables =
+			HashMapBuilder.<String, Object>put(
+				"creator", payloadJSONObject.get("userId")
+			).build();
+
+		Map<String, Object> allVariables = new HashMap<>();
+
+		if (objectDefinition.isSystem()) {
+			String contentType = _getContentType(
+				dtoConverterRegistry, objectDefinition,
+				systemObjectDefinitionMetadataTracker);
+
+			allVariables.putAll(
+				(Map<String, Object>)payloadJSONObject.get(
+					"model" + objectDefinition.getName()));
+			allVariables.putAll(
+				(Map<String, Object>)payloadJSONObject.get(
+					"modelDTO" + contentType));
```

Comment:
Can we `_getContentType` inline?

Reply:
Hi @MarinhoFeliphe, 
I tried doing this but SF recommended creating a variable just like I did